addons/fortify/functions/fnc_updateBudget.sqf

```diff
@@ -0,0 +1,35 @@
+/*
+ * Author: Kingsley
+ * Updates the given sides budget.
+ *
+ * Arguments:
+ * 0: Side <SIDE>
+ * 1: Change <NUMBER> (default: 0)
+ * 2: Display hint <BOOL> (default: true)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [west, -250, false] call acex_fortify_fnc_updateBudget
+ *
+ * Public: Yes
+ */
+
+#include "script_component.hpp"
+
+params [["_side", sideUnknown, [sideUnknown]], ["_change", 0, [0]], ["_hint", true, [true]]];
+TRACE_3("updateBudget",_side,_change,_hint);
+
+if (_side isEqualTo sideUnknown) exitWith {ERROR("Unknown side");};
+
+private _budget = [_side] call FUNC(getBudget);
+private _newBudget = _budget + _change;
+
+if (_budget != -1) then {
+    missionNamespace setVariable [format [QGVAR(Budget_%1), _side], _newBudget, true];
```

Comment:
In theory this isn't multithread/multiplayer safe if 2 players place at the exact same time.
I suppose we might want to move to the server to make it atomic
but it should be very rare and not super critical if one side gets an extra sandbag

Reply:
I guess it might become an issue with very big communities, all doing fortifications at the same time. But then again who cares about one sandbag with that amount of fortifications.